main.tf

```
@@ -308,3 +308,49 @@ resource "aws_codepipeline" "default" {
     }
   }
 }
+
+resource "random_string" "webhook_secret" {
+  count  = local.enabled && var.webhook_enabled ? 1 : 0
+  length = 32
+
+  # Special characters are not allowed in webhook secret (AWS silently ignores webhook callbacks)
+  special = false
+}
+
+locals {
+  webhook_secret = join("", random_string.webhook_secret.*.result)
+  webhook_url    = join("", aws_codepipeline_webhook.webhook.*.url)
+}
+
+resource "aws_codepipeline_webhook" "webhook" {
+  count           = local.enabled && var.webhook_enabled ? 1 : 0
+  name            = module.this.id
+  authentication  = var.webhook_authentication
+  target_action   = var.webhook_target_action
+  target_pipeline = join("", aws_codepipeline.default.*.name)
+
+  authentication_configuration {
+    secret_token = local.webhook_secret
+  }
+
+  filter {
+    json_path    = var.webhook_filter_json_path
+    match_equals = var.webhook_filter_match_equals
+  }
+}
+
+module "github_webhooks" {
+  source  = "cloudposse/repository-webhooks/github"
+  version = "0.12.0"
+
+  enabled              = local.enabled && var.webhook_enabled ? true : false
```

move this to `locals` https://github.com/cloudposse/terraform-aws-cicd/blob/master/main.tf#L7
```hcl
webhook_enabled = local.enabled && var.webhook_enabled ? true : false
```

```suggestion
  enabled              = local.webhook_enabled
```